cfme/tests/configure/test_log_depot_operation.py

```diff
@@ -211,35 +210,56 @@ def depot_configured(request, depot_type, depot_machine_folder, depot_credential
 
     It also provides a finalizer to disable the depot after test run.
     """
-    machine = depot_machine_ip + depot_machine_folder
-    if depot_type not in ["nfs", "anon_ftp"]:
-        credentials = configure.ServerLogDepot.Credentials(
-            depot_type,
-            fauxfactory.gen_alphanumeric(),
-            machine,
-            username=depot_credentials["username"],
-            password=depot_credentials["password"],
-        )
-    else:
-        credentials = configure.ServerLogDepot.Credentials(
-            depot_type,
-            fauxfactory.gen_alphanumeric(),
-            machine,
-        )
-    # Fails on upstream - BZ1108087
-    credentials.update()
-    request.addfinalizer(configure.ServerLogDepot.Credentials.clear)
-    return credentials
+    uri = depot_machine_ip + depot_machine_folder
+    log_depot = configure.ServerLogDepot(depot_type,
+                                         depot_name=fauxfactory.gen_alphanumeric(),
+                                         uri=uri,
+                                         username=depot_credentials["username"],
+                                         password=depot_credentials["password"]
+                                         )
+    log_depot.create()
+    yield log_depot
+
+
+def check_ftp(ftp):
+    with ftp:
+        # Files must have been created after start
+        zip_files = ftp.filesystem.search(re.compile(r"^.*?[.]zip$"), directories=False)
+        assert zip_files, "No logs found!"
+    # Check the times of the files by names
+    datetimes = []
+    regexp = re.compile(
+        r"^.*?_(?P<y1>[0-9]{4})(?P<m1>[0-9]{2})(?P<d1>[0-9]{2})_"
+        r"(?P<h1>[0-9]{2})(?P<M1>[0-9]{2})(?P<S1>[0-9]{2})"
+        r"_(?P<y2>[0-9]{4})(?P<m2>[0-9]{2})(?P<d2>[0-9]{2})_"
+        r"(?P<h2>[0-9]{2})(?P<M2>[0-9]{2})(?P<S2>[0-9]{2})[.]zip$"
+    )
+    for file in zip_files:
+        data = regexp.match(file.name)
+        assert data, "Wrong file matching of {}".format(file.name)
+        data = {key: int(value) for key, value in data.groupdict().iteritems()}
+        date_from = parsetime(
+            data["y1"], data["m1"], data["d1"], data["h1"], data["M1"], data["S1"])
+        date_to = parsetime(data["y2"], data["m2"], data["d2"], data["h2"], data["M2"], data["S2"])
+        datetimes.append((date_from, date_to, file.name))
+
+    # Check for the gaps
+    if len(datetimes) > 1:
+        for i in range(len(datetimes) - 1):
```

Comment:
OPTIONAL: I'd replace this with something like zip(datetimes, datetimes[1:]) or something from itertools/functools.